### PR TITLE
Update to golang 1.18.4

### PR DESCRIPTION
**Pull request by oliver-goetz**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
This PR updates Golang to version 1.18.4
Ref: https://groups.google.com/g/golang-dev/c/frczlF8OFQ0

**Release note**:

```other developer
Golang version is updated to 1.18.4
```

### CODE REVIEW

**Comment by vpnachev** [comment]
It is expected 1.18.4 to be released today, I guess you would like to update to it.
ref: https://groups.google.com/g/golang-nuts/c/gzl4dQ-fQ_A/m/Ph5mkLkOAQAJ

**Review by acumino — COMMENTED** [review]
Just the value need to be changed to 1.18.4 based on the above comment.
/lgtm

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #############      builder       #############
-FROM golang:1.18.1 AS builder
+FROM golang:1.18.3 AS builder
```
Comment (ialidzhikov) [inline Dockerfile]:
```suggestion
FROM golang:1.18.4 AS builder
```

File: .test-defs/CreateShoot.yaml [hunk 2]
```diff
@@ -41,4 +41,4 @@ spec:
 #    -machine-type=$MACHINE_TYPE
 #    -external-domain=
 
-  image: eu.gcr.io/gardener-project/3rd/golang:1.18.1
+  image: eu.gcr.io/gardener-project/3rd/golang:1.18.4
```
Comment (vpnachev) [inline .test-defs/CreateShoot.yaml:44]: This image does not exist. Have you pushed it?

Comment (oliver-goetz) [reply .test-defs/CreateShoot.yaml:44]: It is available now 😄

**Review by ialidzhikov — COMMENTED** [review]
/lgtm

**Pull request merged**